src/overlays/actors/ovl_En_Bom_Bowl_Man/z_en_bom_bowl_man.c

```diff
@@ -39,56 +49,642 @@ const ActorInit En_Bom_Bowl_Man_InitVars = {
+            return;
+        } else if (this->actor.textId == 0x716) {
+            this->actor.textId = 0x735;
+        } else if (this->actor.textId == 0x735) {
+            this->unk_2C2 = 0;
+            func_809C493C(this, 17, 1.0f);
+            func_80151BB4(globalCtx, 0x24);
+            func_80151BB4(globalCtx, 0x25);
+            func_80151BB4(globalCtx, 0);
+            func_800B7298(globalCtx, &this->actor, 7);
+            this->actionFunc = func_809C5738;
+            return;
+        }
+        func_80151938(globalCtx, this->actor.textId);
+    }
+}
+
+void func_809C5738(EnBomBowlMan* this, GlobalContext* globalCtx) {
+    s32 pad[3];
+    s16 yaw = Math_Vec3f_Yaw(&this->actor.world.pos, &this->unk_2A0);
+
+    if (this->unk_2C2 == 0) {
+        if ((globalCtx->msgCtx.unk120B1 == 0) &&
+            ((globalCtx->msgCtx.unk11F22 == 0) || (Message_GetState(&globalCtx->msgCtx) == 6))) {
+            this->unk_2C2 = 1;
+            func_809C4B6C(this);
+            if (ActorCutscene_GetCurrentIndex() == 0x7C) {
+                ActorCutscene_Stop(0x7C);
+                ActorCutscene_SetIntentToPlay(this->unk_2D6);
+            } else if (!ActorCutscene_GetCanPlayNext(this->unk_2D6)) {
+                ActorCutscene_SetIntentToPlay(this->unk_2D6);
+            } else {
+                ActorCutscene_StartAndSetUnkLinkFields(this->unk_2D6, &this->actor);
+                this->unk_2C2 = 2;
+                func_809C493C(this, 18, 1.0f);
+            }
+        }
+    } else if (this->unk_2C2 == 1) {
+        if (ActorCutscene_GetCurrentIndex() == 0x7C) {
+            ActorCutscene_Stop(0x7C);
+            ActorCutscene_SetIntentToPlay(this->unk_2D6);
+        } else if (!ActorCutscene_GetCanPlayNext(this->unk_2D6)) {
+            ActorCutscene_SetIntentToPlay(this->unk_2D6);
+        } else {
+            ActorCutscene_StartAndSetUnkLinkFields(this->unk_2D6, &this->actor);
+            this->unk_2C2 = 2;
+            func_809C493C(this, 18, 1.0f);
+        }
+    } else {
+        Math_SmoothStepToS(&this->actor.world.rot.y, yaw, 1, 2000, 10);
+        Math_ApproachF(&this->actor.world.pos.x, this->unk_2A0.x, 0.5f, 6.0f);
+        Math_ApproachF(&this->actor.world.pos.z, this->unk_2A0.z, 0.5f, 6.0f);
+        if ((sqrtf(SQ(this->actor.world.pos.x - this->unk_2A0.x) + SQ(this->actor.world.pos.z - this->unk_2A0.z)) <
+             4.0f) &&
+            (this->path != NULL)) {
+            this->unk_298++;
+            if (this->unk_298 >= this->path->count) {
+                gSaveContext.weekEventReg[84] |= 0x80;
+                gSaveContext.weekEventReg[83] &= (u8)~4;
+                ActorCutscene_Stop(this->unk_2D6);
+                Actor_MarkForDeath(&this->actor);
+            } else {
+                func_809C4B6C(this);
+            }
+        }
+    }
+}
+
+void func_809C59A4(EnBomBowlMan* this, GlobalContext* globalCtx) {
+    Actor_PickUp(&this->actor, globalCtx, 80, 300.0f, 300.0f);
```

Comment:
Here the third parameter (80) is a value from the GI enum. Taking a quick look at the enum, there doesn't seem to be a value for this. I suggest either asking in the discord which item this could be, or make a placeholder for this value
```suggestion
    Actor_PickUp(&this->actor, globalCtx, 80, 300.0f, 300.0f);
```